src/components/modal/ModalCreateBooks.tsx

```diff
@@ -0,0 +1,188 @@
+import { FormEvent, useEffect, useState } from 'react'
+
+import axios from 'axios'
+import Select, { MultiValue } from 'react-select'
+import ReactDOM from 'react-dom'
+import './ModalCreateBooks.css'
+import { GrClose as XIconForModal } from 'react-icons/gr'
+import { BiBookAdd as BookAddIcon } from 'react-icons/bi'
+
+import BookBody from '../../model/Book'
+import BookService from '../../services/BookService'
+import Author from '../../model/Author'
+import AuthorService from '../../services/AuthorService'
+import placeholder from '../../assets/placeholder/placeholderForBook.png'
+
+
+interface Modal {
+  open: boolean,
+  onClose: () => void
+}
+
+
+const poratlDiv = document.getElementById('portal') as HTMLElement
+const ModalCreateBooks = ({ open , onClose } : Modal) => {
```

Comment:
You should manage modal open / close state outside of the modal itself

Reply:
I fix this.  At now logic for open modal in parrent component.